internal/forge/gitlab/auth.go

```diff
@@ -126,7 +194,22 @@ var _authenticationMethods = []struct {
 	Description func(focused bool) string
 	Build       func(authenticatorOptions) authenticator
 }{
-	// TODO: OAuth
+	{
+		Title:       "OAuth",
+		Description: oauthDesc,
+		Build: func(a authenticatorOptions) authenticator {
+			return &DeviceFlowAuthenticator{
+				ClientID: _oauthAppID,
+				Endpoint: a.Endpoint,
+				Stderr:   a.Stderr,
+				Scopes: []string{
+					"api",
+					"read_user",
+					"write_repository",
+				},
```

Comment:
do we really need all three scopes here?  [GitLab Doc](https://docs.gitlab.com/ee/integration/oauth_provider.html#view-all-authorized-applications)

api | Grants complete read/write access to the API, including all groups and projects, the container registry, the dependency proxy, and the package registry.
-- | --

Reply:
(Long term, r/w on all api is quite broad. I'm hoping there's a narrower set I can put in place here.)